polynote-frontend/polynote/messages.js

```diff
@@ -6,6 +6,27 @@ import {
 } from './codec.js'
 
 import { Result, KernelErrorWithCause } from './result.js'
+import {float64, Pair} from "./codec";
+import {StreamingDataRepr} from "./value_repr";
+
+export function isEqual(a, b) {
+    if (a === b)
+        return true;
+
+    if (a instanceof Array || a instanceof Object) {
+        if (a.constructor !== b.constructor)
+            return false;
+
+        for (const i in a) {
```

Comment:
I wonder whether we'd gain anything by using `Object.entries(a)` instead? For example, would we want a `Map()` and `{}` to be equal if they each have the same kv pairs?

Reply:
That wouldn't work with arrays tho...

Reply:
I think `Object.entries` would work fine (even for arrays), but I guess I don't see the point? And since the constructor is checked, already a `Map` is not going to be equal to an `Object`.

Reply:
Ah, the output is weird for arrays but the equality bit will still work - my bad. 

Didn't think about the constructor check - oops! I think the main question is more like: depending on how this is used, it is possible that we might want two objects to be equal if they have the same keys regardless of their type or whether there might be other keys in their prototypes. In that case, we could use something like `Object.entries`. Otherwise it doesn't matter.